apollo-ios/Sources/ApolloAPI/GraphQLOperation.swift

```diff
@@ -71,10 +84,17 @@ public extension GraphQLOperation {
     return nil
   }
 
-  /// `True` if any selection set, or nested selection set, within the operation contains any
-  /// fragment marked with the `@defer` directive.
-  static var hasDeferredFragments: Bool {
-    false
+  static var deferredFragments: [DeferredFragmentIdentifier: any SelectionSet.Type]? {
+    return nil
+  }
+
+  static func deferredSelectionSetType(
+    withLabel label: String,
+    atPath path: [JSONValue]
```

Comment:
This should take in `[PathComponent]`

Reply:
Done in https://github.com/apollographql/apollo-ios-dev/pull/217/commits/292cc50d4af5bbe997995d5cfa376051e3e65cef.

I changed this to take in `[String]` instead of `[PathComponent]` because `PathComponent` is defined in `Apollo` and `ApolloAPI` has none, and shouldn't have, knowledge of the `Apollo` module. Moving `PathComponent` into `ApolloAPI` didn't seem correct just for this so the logic of filtering out indexes is done by the caller now instead and this function is given a path array of only field names.